xbmc/interfaces/python/PythonInvoker.cpp

```diff
@@ -114,12 +134,6 @@ CPythonInvoker::~CPythonInvoker()
   Stop(true);
   pulseGlobalEvent();
 
-  if (m_argv != NULL)
-  {
-    for (unsigned int i = 0; i < m_argc; i++)
-      delete [] m_argv[i];
-    delete [] m_argv;
-  }
```

Comment:
Replacing the member variables with local ones is the correct fix. However, you replace some overly-complex c++ code with some overly-complicated std container code. I recommend splitting this into two commits, one that moves this block  into CPythonInvoker::execute() and replaces the members with locals (keeping the c++ memory management), and then one that converts from `std::vector<std::string>` to `char**`.